### PR TITLE
 KAFKA-8594: Add version 2.3 to Streams system tests

### PR DESCRIPTION
**Pull request by mjsax**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Triggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2839/

Call for review @guozhangwang @bbejeck @vvcephei @ableegoldman @cadonna @abbccdda

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py [hunk 1]
```diff
@@ -20,7 +20,8 @@
 from kafkatest.services.streams import StreamsBrokerCompatibilityService
 from kafkatest.services.verifiable_consumer import VerifiableConsumer
 from kafkatest.services.zookeeper import ZookeeperService
-from kafkatest.version import LATEST_0_11_0, LATEST_0_10_2, LATEST_0_10_1, LATEST_0_10_0, LATEST_1_0, LATEST_1_1, LATEST_2_0, KafkaVersion
+from kafkatest.version import LATEST_0_11_0, LATEST_0_10_2, LATEST_0_10_1, LATEST_0_10_0, LATEST_1_0, LATEST_1_1, \
+    LATEST_2_0, LATEST_2_1, LATEST_2_2, LATEST_2_3, KafkaVersion
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:24]: Seems we forgot to update this for a couple of releases... If we all agree, I can open corresponding PRs for older branches to close this test gap there, too.

File: tests/kafkatest/tests/streams/streams_multiple_rolling_upgrade_test.py [hunk 2]
```diff
@@ -18,7 +18,8 @@
 from ducktape.mark.resource import cluster
 from kafkatest.services.streams import StreamsSmokeTestDriverService, StreamsSmokeTestJobRunnerService
 from kafkatest.tests.streams.base_streams_test import BaseStreamsTest
-from kafkatest.version import LATEST_0_10_2, LATEST_0_11_0, LATEST_1_0,  DEV_BRANCH, KafkaVersion
+from kafkatest.version import LATEST_0_10_2, LATEST_0_11_0, LATEST_1_0, LATEST_1_1, \
+    LATEST_2_0, LATEST_2_1, LATEST_2_2, LATEST_2_3, DEV_BRANCH, KafkaVersion
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_multiple_rolling_upgrade_test.py:22]: As above...

**Review by ableegoldman — APPROVED** [review]
Thanks Matthias! I started [a list of tests to update](Started [a list of things](https://confluentinc.atlassian.net/wiki/spaces/KSTREAMS/pages/879493954/Streams+Release+Manager+Responsibilities) to update so we can hopefully remember going forward) so we won't forget for several releases going forward...

**Comment by mjsax** [comment]
System test failed. Forgot to update some files. Update this PR. Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2843/

**Comment by ableegoldman** [comment]
System tests passed, Java11/2.13 passed

Java8 & Java11/2.12 failed with unrelated tests. Retest this, please

**Comment by mjsax** [comment]
Rebased to resolve merge conflicts.

Call for review @guozhangwang @bbejeck

File: streams/upgrade-system-tests-23/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
+/*
```
Comment (abbccdda) [inline streams/upgrade-system-tests-23/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java:1]: Why do we need to write a new upgrade test?

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py (hunk 1 above)
Comment (abbccdda) [reply tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:24]: Sounds like a safer approach.

File: streams/upgrade-system-tests-23/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java (hunk 3 above)
Comment (mjsax) [reply streams/upgrade-system-tests-23/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java:1]: Yes. The upgrade system test needs compiled jars for different versions. Hence, we need to add a new module for each version (each module depends on a different---already released---Kafka Streams version).

**Review by bbejeck — APPROVED** [review]
Thanks for adding the new version LGTM

**Pull request merged**